### DISCUSSION TITLE
Need help using api-ring

### DISCUSSION THREAD

**Discussion by multinet33** [dgreif/ring#677] [Q&A]
Hello

I've installed homebridge UI and homebridge ring pluging : all is OK

I've a token

Now I try to interact with my RING alarm using TS bug I get error  when executing sample scripts : 

here is a script I want to run : 

`import 'dotenv/config'
import { RingApi } from 'ring-client-api'

async function example() {
  const { env } = process,
    ringApi = new RingApi({
      // Replace with your refresh token
      refreshToken: "***here is my token****",
    }),
    locations = await ringApi.getLocations(),
    location = locations[0],
    cameras = await ringApi.getCameras(),
    camera = cameras[0]

  // Locations API
  location.onConnected.subscribe((connected) => {
    const state = connected ? 'Connected' : 'Connecting'
    console.log(`${state} to location ${location.name} - ${location.id}`)
  })

  const locationCameraEvents = await location.getCameraEvents({
      // same params as camera.getEvents
    }),
    locationAlarmEvents = await location.getHistory({
      limit: 1,
      category: 'alarm',
      //offset: 100  - number of events to skip over for pagination
    }),
    locationBeamsEvents = await location.getHistory({
      limit: 1,
      category: 'beams',
    })
  console.log('Location Camera Event', locationCameraEvents.events[0])
  console.log('Location Alarm Event', locationAlarmEvents[0])
  console.log('Location Beams Event', locationBeamsEvents[0])

  console.log('Monitoring Status', await location.getAccountMonitoringStatus())

  // Camera API
  const eventsResponse = await camera.getEvents({
    limit: 10,
    kind: 'ding',
    state: 'accepted',
    // olderThanId: previousEventsResponse.meta.pagination_key
    // favorites: true
  })
  console.log('Got events', eventsResponse.events[0])
  const eventsWithRecordings = eventsResponse.events.filter(
      (event) => event.recording_status === 'ready'
    ),
    transcodedUrl = await camera.getRecordingUrl(
      eventsWithRecordings[0].ding_id_str, // MUST use the ding_id_str, not ding_id
      {
        transcoded: true, // get transcoded version of the video.  false by default.  transcoded has ring log and timestamp
      }
    ),
    untranscodedUrl = await camera.getRecordingUrl(
      eventsWithRecordings[0].ding_id_str
    )

  console.log('Recording Transcoded URL', transcodedUrl)
  console.log('Recording Untranscoded URL', untranscodedUrl)
}

example()
`

and that is the error I get : 

root@odroid-stretch64:~# tsc api-example.ts 
node_modules/@homebridge/camera-utils/lib/return-audio-transcoder.d.ts:3:38 - error TS2307: Cannot find module 'homebridge' or its corresponding type declarations.

3 import { PrepareStreamRequest } from 'homebridge';
                                       ~~~~~~~~~~~~

node_modules/ring-client-api/lib/api/location.d.ts:47:30 - error TS2503: Cannot find namespace 'SocketIOClient'.

47     getConnection(): Promise<SocketIOClient.Socket>;
                                ~~~~~~~~~~~~~~

Found 2 errors.

Can you help me 

Thanks a lot

**Comment by dgreif** [comment]
Could you elaborate on your current setup for running the example?  Did you clone the `dgreif/ring` project, or install `ring-client-api` via npm?

**Reply by multinet33** [reply]
Hello, thank you for your reply

I've done an install via npm ring-client-api

My setup : odroid C2 with Linux odroid-stretch64 3.16.85+ #1 SMP PREEMPT Tue Jun 30 19:02:35 CEST 2020 aarch64

I've installed succesfully on it homebridge with the UX interface that I can access wia my browser

On homebridge I've installed successfully the RING plugin for homebrige and i've have a working token. When I open the door, homebridge ring plugin show me that the door is open.

Now I want to arm/disarm my ring alam by script (in order to automate that from domoticz). To do that I tried to use the TS but I get the error I report on my first message.

For the moment I can arm/disarm via a curl command like this : curl -X PUT http://192.168.1.156:51728/characteristics --header "Content-Type:Application/json" --header "authorization: HID-DD-EN" --data "{\"characteristics\":[{\"aid\":2,\"iid\":11,\"value\":1}]}" 

Thanks for you patience and your help :)

Multinet33

**Reply by dgreif** [reply]
I'd recommend using `ts-node api-example.ts` instead of `tsc` if you want to run the example this way.  You can install `ts-node` with `npm i -g ts-node`.  If you continue to have issues, it's probably a lot easier to stick with `curl`ing the homebridge endpoint as you demonstrated

**Reply by dgreif** [reply]
@Leo7Deng check out https://github.com/dgreif/ring-client-example, which has a full example of how to use the client.  You can fork from this example and change `example.ts` to suite your needs
